util/chplenv/chpl_gpu.py

```diff
@@ -13,23 +14,50 @@ def _validate_cuda_version():
 def _validate_rocm_version():
     return _validate_rocm_version_impl()
 
-# Format:
-#   SDK path environment variable
-#   program to locate SDK folder
-#   depth of program within SDK folder
-#   Default GPU architecure for the vendor
-#   LLVM target
+class gpu_type:
```

Comment:
Thank you for taking the initiative to refactor this into a class. Having everything packed as a tuple was starting to get unwieldy.